### PR TITLE
Doc: Cross link to API reference page from Operators page

### LINKED ISSUE #1383: Doc: Cross link to API reference page from Operators page
Currently there is no way to jump to Func/Operator definition from https://astro-sdk-python.readthedocs.io/en/stable/astro/sql/operators/get_value_list.html (and other operators listed on https://astro-sdk-python.readthedocs.io/en/stable/guides/operators.html) to their definition https://astro-sdk-python.readthedocs.io/en/stable/autoapi/index.html

We should cross-link them using https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-syntax

### PR DESCRIPTION
**Pull request by rajaths010494**
Currently there is no way to jump to Func/Operator definition from https://astro-sdk-python.readthedocs.io/en/stable/astro/sql/operators/get_value_list.html (and other operators listed on https://astro-sdk-python.readthedocs.io/en/stable/guides/operators.html) to their definition https://astro-sdk-python.readthedocs.io/en/stable/autoapi/index.html

We should cross-link them using https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-syntax

closes #1383

### CODE REVIEW

File: python-sdk/docs/astro/files/operators/get_file_list.rst [hunk 1]
```diff
@@ -1,8 +1,8 @@
 .. _get_file_list:
 
-=============
-get_file_list
-=============
+=========================================================================
+:py:func:`get_file_list <astro.files.get_file_list>`
+=========================================================================
```
Comment (pankajkoti) [inline python-sdk/docs/astro/files/operators/get_file_list.rst:5]: Can we remove the extra `=======` spanning more than the text content?

File: python-sdk/docs/astro/sql/operators/dataframe.rst [hunk 2]
```diff
@@ -1,8 +1,8 @@
 .. dataframe_operator:
 
-======================================
-dataframe operator
-======================================
+=============================================================
+:py:mod:`dataframe operator <astro.sql.operators.dataframe>`
+=============================================================
```
Comment (pankajkoti) [inline python-sdk/docs/astro/sql/operators/dataframe.rst:6]: Do we have an extra = here too?

File: python-sdk/docs/astro/sql/operators/transform_file.rst [hunk 3]
```diff
@@ -1,8 +1,8 @@
 .. _transform_file_operator:
 
-=======================
-transform_file operator
-=======================
+======================================================================================
+:py:func:`transform_file operator <astro.sql.operators.transform.transform_file>`
+======================================================================================
```
Comment (pankajkoti) [inline python-sdk/docs/astro/sql/operators/transform_file.rst:5]: extra == ?

**Review by pankajkoti — APPROVED** [review]
Looks great, minor suggestions

File: python-sdk/docs/astro/files/operators/get_file_list.rst (hunk 1 above)
Comment (rajaths010494) [reply python-sdk/docs/astro/files/operators/get_file_list.rst]: removed

File: python-sdk/docs/astro/sql/operators/dataframe.rst (hunk 2 above)
Comment (rajaths010494) [reply python-sdk/docs/astro/sql/operators/dataframe.rst]: removed

**Comment by rajaths010494** [comment]
Attaching screenshot for reference 
image

**Review by sunank200 — APPROVED** [review]

**Pull request merged**